x-pack/plugin/search-business-rules/src/yamlRestTest/resources/rest-api-spec/test/search-business-rules/10_pinned_retriever.yml

```diff
@@ -0,0 +1,406 @@
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._id: doc2 }
+
+---
+"pinned retriever as a sub-retriever":
+  - do:
+      search:
+        index: test-index1
+        body:
+          retriever:
+            rrf:
+              retrievers:
+                -
+                  standard:
+                    query:
+                      match_all: {}
+                -
+                  pinned:
+                    ids: ["doc1", "doc2"]
+                    retriever:
+                      standard:
+                        query:
+                          match_all: {}
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+  - match: { hits.hits.2._id: doc3 }
+
+---
+"pinned retriever with explicit sort on score":
+  - do:
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+                  sort: [ "_score" ]
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+  - match: { hits.hits.2._id: doc3 }
+
+---
+"pinned retriever with rank window size":
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+              rank_window_size: 1
+
+  - match: { hits.total.value: 5 }
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._id: doc1 }
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+              rank_window_size: 10
+
+  - match: { hits.total.value: 5 }
+  - length: { hits.hits: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+  - match: { hits.hits.2._id: doc3 }
+
+---
+"pinned retriever explanation":
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+          explain: true
+
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.0._explanation.value: 1.0 }
+  - match: { hits.hits.0._explanation.description: "doc [0] with an original score of [1.7014124E38] is at rank [1] from the following source queries." }
```

Comment:
It would be nice if we called out explicitly that this was pinned in the explanation